### PR TITLE
Upgraded dependencies. Upgraded node. Fixed sneaky bug.

### PR DESCRIPTION
**Pull request by stevieing**
Closes #

Changes proposed in this pull request:

* Upgraded dependencies - including vue, babel, sass, bootstrap, test-utils
* Upgraded to node 16.13.1
* Fixed sneaky bug where I was still using a constructor in plate for replicates as we are now using POJO.

Still seems to be a problem with deprecation warning for SASS.

### CODE REVIEW

File: src/components/Plate.vue [hunk 1]
```diff
@@ -157,7 +156,7 @@ export default {
       } else {
         this.quantType = QuantType()
       }
-      this.replicates = new Replicates(this.quantType.replicateOptions)
+      this.replicates = Replicates(this.quantType.replicateOptions)
```
Comment (KatyTaylor) [inline src/components/Plate.vue:159]: Was this the sneaky bug? What was the symptom?

File: src/components/Upload.vue [hunk 2]
```diff
@@ -88,7 +88,6 @@ export default {
     },
   },
   components: {
-    QuantFile,
```
Comment (KatyTaylor) [inline src/components/Upload.vue:91]: What's the reason for getting rid of this?

**Review by KatyTaylor — APPROVED** [review]
I've approved but left a couple of questions. Some quite big jumps with dependencies but good to get up to date and hopefully I'll be able to test it while doing this story.

**Comment by stevieing** [comment]
Replicates used to be a class. Classes in JavaScript are terrible.

I changed it to an object but was still instantiating it with new so when I upgraded some of the dependencies the tests started failing because it does not have a constructor.

I’m not sure if this would have any effects in production but it could cause subtle bugs.

On 8 Feb 2022, at 13:22, KatyTaylor ***@***.******@***.***>> wrote:



@KatyTaylor commented on this pull request.

________________________________

In src/components/Plate.vue [github.com]<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_sanger_quanthub_pull_148-23discussion-5Fr801623430&d=DwMCaQ&c=D7ByGjS34AllFgecYw0iC6Zq7qlm8uclZFI0SqQnqBo&r=PioAYl_Zz9lnGwJRms8MsEB0E-grNRW-ut-_gAGl70g&m=nzs6Rl9T8lxv93_SYDOOUv9TeyfbJ47cYf8KLO1XbBsEkqWUL9jKfOglhES0j368&s=cFw4096CT2Dayzby-viSOb_1AurMTn4mMs80XMdzBzE&e=>:

> @@ -157,7 +156,7 @@ export default {
       } else {
         this.quantType = QuantType()
       }
-      this.replicates = new Replicates(this.quantType.replicateOptions)
+      this.replicates = Replicates(this.quantType.replicateOptions)


Was this the sneaky bug? What was the symptom?

—
Reply to this email directly, view it on GitHub [github.com]<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_sanger_quanthub_pull_148-23pullrequestreview-2D876015982&d=DwMCaQ&c=D7ByGjS34AllFgecYw0iC6Zq7qlm8uclZFI0SqQnqBo&r=PioAYl_Zz9lnGwJRms8MsEB0E-grNRW-ut-_gAGl70g&m=nzs6Rl9T8lxv93_SYDOOUv9TeyfbJ47cYf8KLO1XbBsEkqWUL9jKfOglhES0j368&s=bGnZY32KpLLT-t7M8oiuxACO4AM7AtHNiw1gLcMQ9qg&e=>, or unsubscribe [github.com]<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AAD6ZH2INUPN77QEWMW22M3U2EKHRANCNFSM5N2H5YHQ&d=DwMCaQ&c=D7ByGjS34AllFgecYw0iC6Zq7qlm8uclZFI0SqQnqBo&r=PioAYl_Zz9lnGwJRms8MsEB0E-grNRW-ut-_gAGl70g&m=nzs6Rl9T8lxv93_SYDOOUv9TeyfbJ47cYf8KLO1XbBsEkqWUL9jKfOglhES0j368&s=JN0NuB4u-XCNSoq4qFkdCup2kkGeS4_gVSzyOXiYzZ0&e=>.
Triage notifications on the go with GitHub Mobile for iOS [apps.apple.com]<https://urldefense.proofpoint.com/v2/url?u=https-3A__apps.apple.com_app_apple-2Dstore_id1477376905-3Fct-3Dnotification-2Demail-26mt-3D8-26pt-3D524675&d=DwMCaQ&c=D7ByGjS34AllFgecYw0iC6Zq7qlm8uclZFI0SqQnqBo&r=PioAYl_Zz9lnGwJRms8MsEB0E-grNRW-ut-_gAGl70g&m=nzs6Rl9T8lxv93_SYDOOUv9TeyfbJ47cYf8KLO1XbBsEkqWUL9jKfOglhES0j368&s=NeYKmIQGD7Txb7exuXVB3MgGXLTFOcOb9qjbeeoJZ8w&e=> or Android [play.google.com]<https://urldefense.proofpoint.com/v2/url?u=https-3A__play.google.com_store_apps_details-3Fid-3Dcom.github.android-26referrer-3Dutm-5Fcampaign-253Dnotification-2Demail-2526utm-5Fmedium-253Demail-2526utm-5Fsource-253Dgithub&d=DwMCaQ&c=D7ByGjS34AllFgecYw0iC6Zq7qlm8uclZFI0SqQnqBo&r=PioAYl_Zz9lnGwJRms8MsEB0E-grNRW-ut-_gAGl70g&m=nzs6Rl9T8lxv93_SYDOOUv9TeyfbJ47cYf8KLO1XbBsEkqWUL9jKfOglhES0j368&s=gz8FZqyYgrTW9caaRMunDK6JkShEf4uDM-e7M2sryZo&e=>.
You are receiving this because you authored the thread.Message ID: ***@***.***>




-- 
 The Wellcome Sanger Institute is operated by Genome Research 
 Limited, a charity registered in England with number 1021457 and a 
 company registered in England with number 2742969, whose registered 
 office is 215 Euston Road, London, NW1 2BE.

**Comment by stevieing** [comment]
It is not used in the tags only in the script so causes a linting error.

QuantFile does not need to be a Vue component so I will probably change it when I get the chance.

On 8 Feb 2022, at 13:22, KatyTaylor ***@***.******@***.***>> wrote:



@KatyTaylor commented on this pull request.

________________________________

In src/components/Upload.vue [github.com]<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_sanger_quanthub_pull_148-23discussion-5Fr801623610&d=DwMCaQ&c=D7ByGjS34AllFgecYw0iC6Zq7qlm8uclZFI0SqQnqBo&r=PioAYl_Zz9lnGwJRms8MsEB0E-grNRW-ut-_gAGl70g&m=3vPjMCwMHN7mZUvMzHhoey4lyt7abSEYmx-yn6kNbbiv0tsIWsXCQ0nG_UrpgpnZ&s=9fOZzpE6IwtgaY5W1ZswSTNj7NyK2hfuv0_mAwM7MZ4&e=>:

> @@ -88,7 +88,6 @@ export default {
     },
   },
   components: {
-    QuantFile,


What's the reason for getting rid of this?

—
Reply to this email directly, view it on GitHub [github.com]<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_sanger_quanthub_pull_148-23pullrequestreview-2D876016271&d=DwMCaQ&c=D7ByGjS34AllFgecYw0iC6Zq7qlm8uclZFI0SqQnqBo&r=PioAYl_Zz9lnGwJRms8MsEB0E-grNRW-ut-_gAGl70g&m=3vPjMCwMHN7mZUvMzHhoey4lyt7abSEYmx-yn6kNbbiv0tsIWsXCQ0nG_UrpgpnZ&s=hgPpI7ANY670zEnyg2UnmoX2JivcO1Wkiuxiyuol6cU&e=>, or unsubscribe [github.com]<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AAD6ZHY22DVWXBDGPU7JLFTU2EKIPANCNFSM5N2H5YHQ&d=DwMCaQ&c=D7ByGjS34AllFgecYw0iC6Zq7qlm8uclZFI0SqQnqBo&r=PioAYl_Zz9lnGwJRms8MsEB0E-grNRW-ut-_gAGl70g&m=3vPjMCwMHN7mZUvMzHhoey4lyt7abSEYmx-yn6kNbbiv0tsIWsXCQ0nG_UrpgpnZ&s=Oyj_AisE0I83Swd28fi8KBShkobL57pD1bKIJ8JISn8&e=>.
Triage notifications on the go with GitHub Mobile for iOS [apps.apple.com]<https://urldefense.proofpoint.com/v2/url?u=https-3A__apps.apple.com_app_apple-2Dstore_id1477376905-3Fct-3Dnotification-2Demail-26mt-3D8-26pt-3D524675&d=DwMCaQ&c=D7ByGjS34AllFgecYw0iC6Zq7qlm8uclZFI0SqQnqBo&r=PioAYl_Zz9lnGwJRms8MsEB0E-grNRW-ut-_gAGl70g&m=3vPjMCwMHN7mZUvMzHhoey4lyt7abSEYmx-yn6kNbbiv0tsIWsXCQ0nG_UrpgpnZ&s=nXxlOYHIucLQfL_GVG6hG053IR0teYxytTYWB5y_EBs&e=> or Android [play.google.com]<https://urldefense.proofpoint.com/v2/url?u=https-3A__play.google.com_store_apps_details-3Fid-3Dcom.github.android-26referrer-3Dutm-5Fcampaign-253Dnotification-2Demail-2526utm-5Fmedium-253Demail-2526utm-5Fsource-253Dgithub&d=DwMCaQ&c=D7ByGjS34AllFgecYw0iC6Zq7qlm8uclZFI0SqQnqBo&r=PioAYl_Zz9lnGwJRms8MsEB0E-grNRW-ut-_gAGl70g&m=3vPjMCwMHN7mZUvMzHhoey4lyt7abSEYmx-yn6kNbbiv0tsIWsXCQ0nG_UrpgpnZ&s=dkm9zIitqps0xSkYvTVkQMg3J1QbDcIfVUAuUhiVx_g&e=>.
You are receiving this because you authored the thread.Message ID: ***@***.***>




-- 
 The Wellcome Sanger Institute is operated by Genome Research 
 Limited, a charity registered in England with number 1021457 and a 
 company registered in England with number 2742969, whose registered 
 office is 215 Euston Road, London, NW1 2BE.

**Pull request merged**